### PR TITLE
No longer kills people who become monkeys. Oops.

### LINKED ISSUE #26641: Deleting your mob teleports your ghost to the arrivals shuttle
Caused by [12406125a8441ccf7787c26c84a3e7d2d8e5fa75] Braining/debraining fixes (#26346).

@AnturK

### PR DESCRIPTION
**Pull request by QualityVan**
:panicbasket:
seriously though why did the new body qdel their organs when insertion and removal was handled fine in the rest of the proc

Has been more tested now, fixes all issues related to #26346 I'm aware of including:
Dummies with godmode dying when their brain is destroyed
Changelings not being able to revive without a brain(turns out the carbon revive() specifically checks for one)
People dying when transforming between monkeys and humans.
Ghosts teleporting to the arrivals shuttle if their mob is deleted, fixes #26641

### CODE REVIEW

File: code/modules/mob/living/brain/brain_item.dm [hunk 1]
```diff
@@ -60,6 +60,8 @@
 	name = "[L.name]'s brain"
 	if(brainmob || decoy_override)
 		return
+	if(L.mind.current != L)
```
Comment (optimumtact) [inline code/modules/mob/living/brain/brain_item.dm]: what is this

Comment (QualityVan) [inline code/modules/mob/living/brain/brain_item.dm]: Prevents the creation of a brainmob when the body's mind has already been transferred to a new mob.

Comment (QualityVan) [inline code/modules/mob/living/brain/brain_item.dm]: Inserting a brain into a body will kick whoever's in the body out, but only if the brain is occupied. This prevents brains from being occupied when they shouldn't be.

Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/brain/brain_item.dm]: this definitely hurts my brain...

Comment (QualityVan) [inline code/modules/mob/living/brain/brain_item.dm]: Turns out I was fixing something in a really stupid way. Will revise.

**Comment by QualityVan** [comment]
Atomizing fixes into multiple PRs.

**Pull request closed without merging**